### PR TITLE
New Integration - Ipcalc

### PR DESCRIPTION
**Pull request by ayman-m**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Description
An integration pack to help automate subnet calculations

## Minimum version of Cortex XSOAR
- [x ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x ] No

## Must have
- [ ] Tests
- [x ] Documentation

### CODE REVIEW

**Comment by ayman-m** [comment]
The docker image PR: 6350

File: Packs/NetCalc/Integrations/IPCalc/IPCalc.py [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+    ipcalc.Network('192.168.10.10/32')
+    return 'ok'
+
+
+def return_subnet_addresses_command(args: Dict[str, Any]) -> CommandResults:
+
+    subnet = args.get('subnet', None)
+
+    if not subnet:
+        raise ValueError('subnet not specified')
```
Comment (daryakoval) [inline Packs/NetCalc/Integrations/IPCalc/IPCalc.py:23]: If `subnet` arg is mandatory, no need to do this check.  (Suggestion for next integrations, no need to apply it now).

**Review by daryakoval — CHANGES_REQUESTED** [review]

**Comment by daryakoval** [comment]
Pleae add to `.secrets-ignore` file all these ip addresses: 
```
192.168.20.21
192.168.20.22
192.168.16.1
192.168.23.254
192.168.23.255
192.168.20.20
```

**Comment by daryakoval** [comment]
@ayman-m Hi, Your integration is not really integrates with some product using product's API. 
Consider to create scripts for each command instead of integration. (Not mandatory change, just suggestion).

File: Packs/NetCalc/Integrations/IPCalc/IPCalc.yml [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    execution: false
+    name: ipcalc-return-subnet-binary
+    outputs:
+    - contextPath: IPCalc.IP.Binary.binary
+      description: Subnet binary
+      type: String
+    - contextPath: IPCalc.IP.Binary.subnet
+      description: Subnet address
+      type: String
+  dockerimage: aymanam/ipcalc:latest
```
[the remark this answers is not in the document]
Comment (ayman-m) [reply Packs/NetCalc/Integrations/IPCalc/IPCalc.yml:155]: yes, PR 6350

File: Packs/NetCalc/Integrations/IPCalc/IPCalc.py (hunk 1 above)
Comment (ayman-m) [reply Packs/NetCalc/Integrations/IPCalc/IPCalc.py:23]: sure

**Comment by ayman-m** [comment]
> @ayman-m Hi, Your integration is not really integrates with some product using product's API. Consider to create scripts for each command instead of integration. (Not mandatory change, just suggestion).

this was actually my initial plan but I thought too many scripts , will change it to automation scripts

**Comment by daryakoval** [comment]
https://github.com/demisto/dockerfiles/pull/6350 Docker file

**Comment by daryakoval** [comment]
@ayman-m Please check if this docker image works: `devdemisto/ipcalc:1.0.0.24320`. 
If it is ok, update me and we will merge docker image pr.

**Review by daryakoval — CHANGES_REQUESTED** [review]
Why `Packs/NetCalc/Scripts/IPCalcReturnSubnetLastAddress/README.md` file is empty?

**Comment by ayman-m** [comment]
> Why `Packs/NetCalc/Scripts/IPCalcReturnSubnetLastAddress/README.md` file is empty?

I fixed that

**Comment by ayman-m** [comment]
> @ayman-m Please check if this docker image works: `devdemisto/ipcalc:1.0.0.24320`. If it is ok, update me and we will merge docker image pr.
I am not able to pull this image, it doesn't exist

**Comment by ayman-m** [comment]
@daryakoval I did update the scripts to use the ipaddress library , no need for a new docker image, so I closed the docker pr

**Review by daryakoval — APPROVED** [review]

**Pull request merged**